src/java/com/sarality/app/view/fragment/BaseFragment.java

```diff
@@ -19,54 +18,95 @@
  */
 public abstract class BaseFragment<V extends View, T> extends Fragment implements DataConsumer<T> {
 
+  private DataSource<T> dataSource;
 
-  private final FragmentActivity activity;
-  private final int viewResourceId;
-  private final ViewRenderer<V, T> renderer;
-  private final DataSource<T> dataSource;
-
-  public BaseFragment(FragmentActivity activity, int viewResourceId, ViewRenderer<V, T> renderer,
-      DataSource<T> dataSource) {
-    super();
-    this.activity =activity;
-    this.viewResourceId = viewResourceId;
-    this.renderer = renderer;
-    this.dataSource = dataSource;
+  @Override
+  public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+    // Inflate the view identified by the resource id as the view for the fragment.
+    return inflater.inflate(getLayoutResourceId(), null, false);
   }
 
   @Override
-  @SuppressWarnings("unchecked")
-  public V getView() {
-    return (V) super.getView();
+  public void onActivityCreated(Bundle savedInstanceState) {
+    super.onActivityCreated(savedInstanceState);
+    dataSource = createDataSource();
+    startLoad(dataSource, getLoaderId(), this);
   }
 
   /**
-   * @return Fragment activity that instantiated the fragment.
+   * @return Resource Id for the View used to render the Fragment.
+   */
+  protected abstract int getLayoutResourceId();
+
+  /**
+   * @return View Id of the View that is rendered by the Renderer. Return {@code null} when a renderer is not defined.
    */
-  protected FragmentActivity getFragmentActivity() {
-    return activity;
+  protected abstract int getRendererViewId();
+
+  protected V getRendererView() {
+    @SuppressWarnings("unchecked")
+    V view = (V) getView().findViewById(getRendererViewId());
+    return view;
   }
 
-  @Override
-  public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-    // Inflate the view identified by the resource id as the view for the fragment.
-    return inflater.inflate(viewResourceId, null, false);
+  /**
+   * Creates a data source used to load the data for the fragment.
+   *
+   * @return The DataSource used to load data for the fragment.
+   */
+  protected abstract DataSource<T> createDataSource();
+
+  /**
+   * Returns a unique Id to be used for the loader that loads the data from the data source.
+   * <p/>
+   * The fragment MUST override this method if there are more than one loader in the activity. Returns 0 by default
+   * with the assumption that there is only one loader being used in the activity.
+   *
+   * @return Unique Id for the loader.
+   */
+  protected int getLoaderId() {
+    return 0;
```

Comment:
i was suggesting that we just print a message so that whoever is writing the app using these initialisers know what he/she is doing.  I agree that forcing them to use implement getLoaderId may be too much.